client/src/config.rs

```diff
@@ -78,8 +78,9 @@ impl Config {
     /// - unable to load config from a TOML file
     /// - the config is invalid
     pub fn load(path: impl AsRef<Path>) -> error_stack::Result<Self, LoadError> {
+        let toml = TomlSource::from_file(path).change_context(LoadError)?;
         let config = ConfigReader::new()
-            .with_toml_source(TomlSource::from_file(path).change_context(LoadError)?)
+            .with_toml_source(toml)
             .read_and_complete::<user::Root>()
             .change_context(LoadError)?
```

Comment:
what if we get an error here? Won't this also fail with drop bomb?

Reply:
shouldn't since the `ConfigReader` wouldn't have been constructed yet

Reply:
> what if we get an error here? Won't this also fail with drop bomb?

`read_and_complete` defuses the bomb, so no, it shouldn't. I _believe_ tests cover it.